### PR TITLE
Feat/75 improve theme switcher

### LINKED ISSUE #75: Theme switcher of code view is confusing
During the CSSclasses today in Hamburg we got some feedback about the usability of the materials page, specifically the code view and its theme switcher.

## Current behaviour
- First: Non-Coders are not used to dark/light themes or mostly have no preference. So the prominentely placed switcher may confuse ("Why dark/light, whats 'better'?")
- Second: The activated state of the theme switcher (for light theme a white dot in red box) looks more like the learned "empty checkbox". So a click on the assumed "unactive" element has no effect so it seems broken or useless and leads to the feeling of "Am I doing something wrong?"

theme-switcher

## Proposed behaviour
Either the checked state should be inverted or made clearer (hook, dot etc.)

What do you think?

### PR DESCRIPTION
**Pull request by Dangoo**
This PR fixes #75 and improves the usability and behavior of the theme switch for code panes.
The refactoring includes:
- extracting of the switch markup as include
- fixing the existence of not unique ids for switch inputs (`dark`, `light` ) by adding a counter
- optimized  syncing of switches
- improved design of switch and improved highlight of active state

If theres something missing, just let me know :)

### CODE REVIEW

**Comment by verpixelt** [comment]
Will have a look later today. Thanks for the PR =)

File: _sass/modules/_m-switch.scss [hunk 1]
```diff
@@ -1,36 +1,75 @@
...
+		height: $switcher-height;
+		border-radius: $switcher-height / 2;
+		background-color: $input-bg-color;
+    transform: translateX(-50%);
+  }  
 
-.m-switch.is--dark .m-switch__label--is-dark:before {
-  border: .25rem solid $color-persian-red;
-  background-color: $color-mine-shaft;
+  &__input {
```
Comment (verpixelt) [inline _sass/modules/_m-switch.scss]: Please get rid of the nesting, except for pseudo elements and pseudo classes.

File: _sass/modules/_m-switch.scss [hunk 2]
```diff
@@ -1,36 +1,75 @@
...
+  $thumb-color: $color-white;
+  $input-bg-color: $color-persian-red;
+  $label-hl-color: $color-persian-red;
+  
+	position: relative;
+	display: inline-flex;
+	margin: $switcher-outer-spacing 0;
+	user-select: none;
+  -webkit-tap-highlight-color: rgba(0,0,0,0);
+	&__label {
```
Comment (verpixelt) [inline _sass/modules/_m-switch.scss]: Please make the labels sans-serif

**Comment by Dangoo** [comment]
@verpixelt If I had known you would be reviewing this, I wouldn't have used nested selectors in the first place. Nice to see some things never change 😂 

PR updated

**Comment by Dangoo** [comment]
@verpixelt @lukaszklis Any chance on merging and deploying that prior to tomorrow’s CSSclasses in HH?

**Comment by verpixelt** [comment]
I'll have a closer look later today. Should be possible to merge it today.

**Comment by jonicious** [comment]
Thanks for the work @Dangoo! After some trouble with our deployment ( ¯\_(ツ)_/¯), the new theme switcher is live! 🎉

**Comment by Dangoo** [comment]
❤️

**Pull request merged**